packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,707 @@
+                            #endif
+                            if (grlColorMode == ${GreasedLineMeshColorMode.COLOR_MODE_SET}.) {
+                                gl_FragColor = grlColor;
+                            } else if (grlColorMode == ${GreasedLineMeshColorMode.COLOR_MODE_ADD}.) {
+                                gl_FragColor += grlColor;
+                            } else if (grlColorMode == ${GreasedLineMeshColorMode.COLOR_MODE_MULTIPLY}.) {
+                                gl_FragColor *= grlColor;
+                            }
+                        }
+                    #endif
+                `,
+            };
+        }
+
+        return null;
+    }
+
+    /**
+     * Converts boolean to number.
+     * @param bool
+     * @returns 1 if true, 0 if false.
+     */
+    private static _BooleanToNumber(bool?: boolean) {
+        return bool ? 1 : 0;
+    }
+
+    /**
+     * Converts an array of Color3 to Uint8Array
+     * @param colors Arrray of Color3
+     * @returns Uin8Array of colors [r, g, b, a, r, g, b, a, ...]
+     */
+    private static _Color3toRGBAUint8(colors: Color3[]) {
+        const colorTable: Uint8Array = new Uint8Array(colors.length * 4);
+        for (let i = 0, j = 0; i < colors.length; i++) {
+            colorTable[j++] = colors[i].r * 255;
+            colorTable[j++] = colors[i].g * 255;
+            colorTable[j++] = colors[i].b * 255;
+            colorTable[j++] = 255;
+        }
+
+        return colorTable;
+    }
+
+    /**
+     * Creates a RawTexture from an RGBA color array and sets it on the plugin material instance.
+     * @param name name of the texture
+     * @param colors Uint8Array of colors
+     */
+    private _createColorsTexture(name: string, colors: Color3[]) {
+        const colorsArray = GreasedLinePluginMaterial._Color3toRGBAUint8(colors);
+        this._colorsTexture = new RawTexture(
+            colorsArray,
+            colors.length,
+            1,
+            Engine.TEXTUREFORMAT_RGBA,
+            this._scene,
+            false,
+            true,
+            this._options.colorsSampling ?? RawTexture.NEAREST_NEAREST
+        );
+        this._colorsTexture.name = name;
+    }
+
+    /**
+     * Disposes the plugin material.
+     */
+    public dispose(): void {
+        this._colorsTexture?.dispose();
+        super.dispose();
+    }
+
+    /**
+     * Sets whether to use the colors option to colorize the line.
+     * @param value true if use the colors, otherwise false
+     */
+    public setUseColors(value: boolean) {
+        this._options.useColors = value;
+    }
+
+    /**
+     * Creates or updates the colors texture
+     * @param colors color table RGBA
+     * @param lazy if lazy, the colors are not updated
+     * @param forceUpdate force creation of a new texture
+     * @returns
+     */
+    public setColors(colors: Nullable<Color3[]>, lazy = false, forceUpdate = false): void {
+        if (colors === null || colors.length === 0) {

```

Comment:
Is this called frequently ? and if yes, mostly with the same array length ?

Reply:
This is called once when creating the line instance and everytime a new line is added to the instance. So it can be called quite frequently when building in instance mode. The array length changes get bigger everytime a new line is added to the instnace.